cmd/govim/internal/fswatcher/os_darwin.go

```diff
@@ -7,19 +7,29 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"sync"
 	"time"
 
 	"github.com/fsnotify/fsevents"
 	"gopkg.in/tomb.v2"
 )
 
-const fRemoved = fsevents.ItemRemoved | fsevents.ItemRenamed
-const fChanged = fsevents.ItemModified | fsevents.ItemChangeOwner
-const fCreated = fsevents.ItemCreated
+const (
+	fRemoved = fsevents.ItemRemoved | fsevents.ItemRenamed
+	fChanged = fsevents.ItemModified | fsevents.ItemChangeOwner
+	fCreated = fsevents.ItemCreated
+	fIsDir   = fsevents.ItemIsDir
+)
 
 type fswatcher struct {
 	eventCh chan Event
 	es      *fsevents.EventStream
+
+	// Darwin do recursive watching so we need to filter files in directories that
+	// wasn't explicitly added. Note that it is desirable to watch recursively to avoid
+	// a data race (#492).
+	watched     map[string]bool // keyed by full path to directory
```

Comment:
Per our offline chant, I'm not clear why this is necessary. On Darwin, given the watcher _is_ recursive (which is a good thing) we don't need to do the `filepath.Walk` that adds recursive watchers. 

Instead, we can simply check on the event callback whether the event is for a file we care about. i.e. ignore `.` and `_` prefixed directories, as well as submodules. 

It's not perfect of course and there are loads of potential FS races... but that's the nature of the beast 😄